src/tests/cli_tests.py

```diff
@@ -673,219 +542,408 @@ def rnp_cleartext_signing_gpg_to_rnp(filesize):
+            random_text(src_beg, 1000)
+
+            run_proc(RNP, ['--enarmor', data_type, src_beg, '--output', dst_beg])
+            run_proc(RNP, ['--dearmor', dst_beg, '--output', dst_mid])
+            run_proc(RNP, ['--enarmor', data_type, dst_mid, '--output', dst_fin])
+
+            compare_files(dst_beg, dst_fin, "RNP armor/dearmor test failed")
+            compare_files(src_beg, dst_mid, "RNP armor/dearmor test failed")
+            remove_files(dst_beg, dst_mid, dst_fin)
+
+class Encryption(unittest.TestCase):
+    '''
+        Things to try later:
+        - different public key algorithms
+        - different hash algorithms where applicable
+        - cleartext signing/verification
+        - detached signing/verification
+
+        TODO:
+        Tests in this test case should be splitted into many algorithm-specific tests (potentially auto generated)
+        Reason being - if you have a problem with BLOWFISH size 1000000, you don't want to wait until everything else gets
+        tested before your failing BLOWFISH
+    '''
+    RNP_CIPHERS = ['AES', 'AES192', 'AES256', 'TWOFISH', 'CAMELLIA128', 'CAMELLIA192', 'CAMELLIA256', 'IDEA', '3DES', 'CAST5', 'BLOWFISH']
+    GPG_CIPHERS = ['cast5', 'idea', 'blowfish', 'twofish', 'aes128', 'aes192', 'aes256', 'camellia128', 'camellia192', 'camellia256', 'tripledes']
+    SIZES = [20, 1000, 5000, 20000, 150000, 1000000]
+
+    @classmethod
+    def setUpClass(cls):
+        # Generate keypair in RNP
+        rnp_genkey_rsa(KEY_ENCRYPT)
+        # Add some other keys to the keyring
+        rnp_genkey_rsa('dummy1@rnp', 1024)
+        rnp_genkey_rsa('dummy2@rnp', 1024)
+        gpg_import_pubring()
+        gpg_import_secring()
+
+    @classmethod
+    def tearDownClass(cls):
+        clear_keyrings()
 
-'''
-    Things to try here later on:
-    - different public key algorithms
-    - different key protection levels/algorithms
-    - armored import/export
-'''
-def run_rnpkeys_tests():
-    # 1. Generate default RSA key
-    run_test(rnpkey_generate_rsa)
-    # 2. Generate 4096-bit RSA key
-    run_test(rnpkey_generate_rsa, 4096)
-    # 3. Generate multiple RSA keys and check if they are all available
-    run_test(rnpkey_generate_multiple)
-    # 4. Generate key with GnuPG and import it to rnp
-    run_test(rnpkey_import_from_gpg)
-    # 5. Generate key with RNP and export it and then import to GnuPG
-    run_test(rnpkey_export_to_gpg)
-
-def run_tests():
-    global DEBUG
-
-    # Parsing command line parameters
-    try:
-        opts, args = getopt.getopt(sys.argv, 'hd', ['help', 'debug'])
-    except getopt.GetoptError:
-        print "Wrong usage. Run cli_tests --help"
-        sys.exit(2)
-
-    tests = []
-
-    for arg in args[1:]:
-        if arg in ['-h', '--help']:
-            print 'Usage:\ncli_tests [-h | --help] [rnp] [rnpkeys] [all] [-d | --debug]'
-            sys.exit(0)
-        elif arg in ['-d', '--debug']:
-            DEBUG = True
```

Comment:
Looks like we lost ability to print out all running commands, at least it doesn't work in `-v` mode

Reply:
Yeah I noticed this too, I think it was very useful. Maybe we should count the `-v` args and have a verbosity level? (Or bring the debug arg back for this?)

Reply:
Yes, the output is different. ``-v`` shows all the tests being run. Unfortunately those tests needs to be further redesigned. Currently in one single test runs f.e. encryption for all algorithms, sizes, compressions... this is far from useful.

Reply:
hmmm... which interesting output you are talking about exactly?

Reply:
Done - now we are using ``logging`` module to get debug logs. ``-d`` option is back